pkg/compose/run.go

```diff
@@ -89,7 +89,7 @@ func (s *composeService) prepareRun(ctx context.Context, project *types.Project,
 	updateServices(&service, observedState)
 
 	created, err := s.createContainer(ctx, project, service, service.ContainerName, 1,
-		opts.Detach && opts.AutoRemove, opts.UseNetworkAliases, opts.Interactive)
```

Comment:
Was this because there was some code on the compose side to handle this case? (that has been removed?) 

Just double-checking if there's not also code on compose's side, causing a potential race on "who gets to remove it first" 😂

Reply:
Yep it was, previously the remove part was done by Compose, [see here](https://github.com/docker/compose/commit/1d4b4e3c8ebd60c2d3f71cb8f26801fee43b8b3f)

```
func (s *composeService) terminateRun(ctx context.Context, containerID string, opts api.RunOptions) (exitCode int, err error) {
	exitCh, errCh := s.apiClient().ContainerWait(ctx, containerID, container.WaitConditionNotRunning)
	select {
	case exit := <-exitCh:
		exitCode = int(exit.StatusCode)
	case err = <-errCh:
		return
	}
	if opts.AutoRemove {
		err = s.apiClient().ContainerRemove(ctx, containerID, moby.ContainerRemoveOptions{})
	}
	return
}
```

Reply:
Ah, great! Thanks for confirming 👍